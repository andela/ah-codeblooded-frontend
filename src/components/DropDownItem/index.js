import React from 'react';
import PropTypes from 'prop-types';

const DropDownItem = ({ link, subList, children }) => (
  <li>
    {
      subList ? (
        <ul>
          {subList}
        </ul>
      )
        : (
          <a href={link}>{children}</a>
        )
    }
  </li>
);


DropDownItem.defaultProps = {
  link: null,
  subList: null,
  children: null,
};

DropDownItem.propTypes = {
  link: PropTypes.string,
  subList: PropTypes.arrayOf(PropTypes.shape()),
  children: PropTypes.oneOfType([PropTypes.string, PropTypes.shape]),
};
export default DropDownItem;
