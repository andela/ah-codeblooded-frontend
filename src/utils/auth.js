export const getCurrentUser = () => {
  // read the user details from local storage
  const user = localStorage.getItem('user');
  return user ? JSON.parse(user) : null;
};
export const getToken = () => (getCurrentUser() ? getCurrentUser().token : null);
