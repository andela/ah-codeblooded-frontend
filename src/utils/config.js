const config = {
  BASE_URL: process.env.API_URL,
};

export default config;
