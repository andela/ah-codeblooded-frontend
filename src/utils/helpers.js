export const randomKey = () => (
  Math.floor((Math.random() * 100) + 1)
);

export const getQueryParam = (param) => {
  const urlParams = new URL(window.location.href).searchParams;
  return urlParams.get(param);
};


export const getLocation = () => window.location.toString();

export const reloadLocation = () => window.location.assign(window.location.href);
