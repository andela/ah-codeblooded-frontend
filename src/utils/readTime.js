export default (seconds) => {
  if (seconds < 60) {
    const s = Math.round(seconds);
    return `${s + (s === 1 ? ' Second' : ' Seconds')} read`;
  } if (seconds < 3600) {
    const minutes = Math.round(seconds / 60);
    return `${minutes + (minutes === 1 ? ' Minute' : ' Minutes')} read`;
  }
  return 'More than 1 hour';
};
