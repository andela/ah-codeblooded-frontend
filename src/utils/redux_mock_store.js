import configureMockStore from 'redux-mock-store';

import { middlewares } from '../store';

const mockStore = configureMockStore(middlewares);

export default mockStore;
