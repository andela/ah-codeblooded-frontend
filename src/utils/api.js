import axios from 'axios';
import config from './config';
import { getToken } from './auth';
import store from '../store';
import { NETWORK_ERROR_DETECTED, NO_NETWORK_ERROR } from "../containers/NetworkPopup/state/types";

const token = getToken();

const uri = config.BASE_URL;

export const getURL = endpoint => `${uri}${endpoint}`;

const api = axios.create({
  baseURL: uri,
});

api.interceptors.request.use(
  (cfg) => {
    if (token) {
      cfg.headers.authorization = `Token ${token}`;
    }
    return cfg;
  },
);

api.interceptors.response.use(
  (response) => {
    store.dispatch({ type: NO_NETWORK_ERROR });
    return response;
  },
  (error) => {
    if (!error.response) {
      store.dispatch({ type: NETWORK_ERROR_DETECTED });
    }
    return Promise.reject(error);
  },
);

api.interceptors.response.use(
  response => response,
  (error) => {
    if (error.response && error.response.status === 403) {
      localStorage.setItem('user', null);
      window.location.assign(`${window.location.protocol}//${window.location.host.toString()}/login`);
    }
    return Promise.reject(error);
  },
);
export default api;
