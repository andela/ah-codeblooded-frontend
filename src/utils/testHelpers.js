import { shallow } from 'enzyme';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import MockAdapter from 'axios-mock-adapter';
import api from './api';
import editorState from '../pages/Articles/Create/state/editorState';
import { initialState } from '../pages/Articles/state/reducer';

const middlewares = [thunk];
export const mockStore = configureMockStore(middlewares);
export const axiosMock = new MockAdapter(api);

export const testReducer = (action, value, result, reducer, state) => {
  const newState = reducer(state, action);
  expect(newState[value])
    .toEqual(result);
};

export const user = {
  username: 'gitaumoses4',
  image: 'https://avatars3.githubusercontent.com/u/25629064?s=460&v=4',
  email: 'gitaumoses4@gmail.com',
  password: 'Password1!',
};

export const article = {
  title: 'The Dungeon Dragon',
  body: JSON.stringify(editorState),
  description: 'The Dungeon Dragon Story',
  slug: 'the-dungeon-dragon-12jhgf323',
  tags: ['dragons', 'dungeon'],
  published: false,
  author: user,
};

export const comment = {
  id: 8,
  body: "Here we go",
  author: {
    username: "gitaumoses4",
    bio: "This is my story, this is my song. Praising my saviour all thy day long",
    image: "https://res.cloudinary.com/codeblooded/image/upload/v1542154278/lwwun6s61usi3jnxjulz.jpg",
  },
  likes: {
    count: 0,
    me: false,
  },
  dislikes: {
    count: 0,
    me: false,
  },
  parent: null,
  created_at: "2018-11-21T23:41:34.216533Z",
};

export const articleCRUD = {
  ...initialState,
};


export const checkSnapshot = (component) => {
  const snap = shallow(component);
  expect(snap).toMatchSnapshot();
};

export const loginUser = (usr = user) => {
  localStorage.setItem("user", JSON.stringify(usr));
};
