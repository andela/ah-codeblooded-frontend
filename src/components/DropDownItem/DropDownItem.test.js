import React from 'react';
import { checkSnapshot } from '../../utils/testHelpers';
import DropDownItem from './index';

describe('<DropDown> component render', () => {
  it('should render a <DropDownItem> component without crashing', () => {
    checkSnapshot(<DropDownItem />);
  });
});
