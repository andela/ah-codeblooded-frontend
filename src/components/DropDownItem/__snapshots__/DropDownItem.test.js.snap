// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DropDown> component render should render a <DropDownItem> component without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DropDownItem
    classNames=""
    link={null}
    onClick={null}
    subList={null}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <a
        href={null}
      />,
      "className": "dropdown-item ",
      "onClick": null,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": null,
        "href": null,
      },
      "ref": null,
      "rendered": null,
      "type": "a",
    },
    "type": "li",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <a
          href={null}
        />,
        "className": "dropdown-item ",
        "onClick": null,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": null,
          "href": null,
        },
        "ref": null,
        "rendered": null,
        "type": "a",
      },
      "type": "li",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
